utils/rust-setup.js

```diff
@@ -0,0 +1,109 @@
+    console.log(chalk`Checking if {bold rustup} is installed...`);
+    const isInstalled = sh.exec('rustup --version &> /dev/null').code == 0;
+    console.log(chalk`{bold rustup} is`, isInstalled ? 'installed' : 'not installed');
+    return isInstalled;
+}
+
+function isWasmTargetAdded() {
+    const addedTargets = sh.exec('rustup target list --installed').stdout;
+    const isWasmTargetAdded = addedTargets.includes('wasm32-unknown-unknown');
+    console.log(chalk`{bold wasm32-unknown-unknown} target `, isWasmTargetAdded ? 'already added' : 'is not added');
+    return isWasmTargetAdded;
+}
+
+function installRustup() {
+    console.log(chalk`Installing {bold rustup}...`);
+    sh.exec(installRustupScript);
+}
+
+function addWasm32Target() {
+    console.log(chalk`Adding {bold wasm32-unknown-unknown} target...`);
+    sh.exec(updatePathAndAddWasm32TargetScript);
+}
+
+function askYesNoQuestionAndRunFunction(question, functionToRun = null) {
+    for (let attempt = 0; attempt < 4; attempt++) {
+        const answer = reader.question(question);
+        if (answer.toLowerCase() == 'y' || answer === '') {
+            if (functionToRun) functionToRun();
+            return true;
+        }
+        else if (answer.toLowerCase() == 'n') {
+            return false;
+        }
+    }
+    return false;
+}
+
+const installRustupDisclaimer = chalk`In order to work with {bold rust} smart contracts we recomend you to install {bold rustup}: the Rust toolchain installer.`;
+const addWasm32TargetDisclaimer = chalk`To build Rust smart contracts you need to add {bold WebAssembly} compiler target to Rust toolchain.`;
+
+const installRustupQuestion = chalk`
+${installRustupDisclaimer}
+We can run the following command to do it:
+
+    {bold ${installRustupScript}}
+
+Continue with installation (y/n)?: `;
+
+const addWasm32TragetQuestion = chalk`
+${addWasm32TargetDisclaimer}
+We can run the following command to do it:
+
+    {bold ${updatePathAndAddWasm32TargetScript}}
+
+Continue with installation (y/n)?:`;
+
+const rustupAndWasm32WindowsInstalationInstructions = chalk`
+${installRustupDisclaimer}
+    1. Go to https://rustup.rs
+    2. Download {bold rustup-init.exe}
+    3. Install it on your system
+
+${addWasm32TargetDisclaimer}
+Run the following command to do it:
+
+    {bold ${addWasm32TargetScript}}
+
+Press {bold Enter} to continue project creation.`;
+
+function setupRustAndWasm32Target() {
+    try {
+        if (isUnix) {
+            if (isRustupInstalled()) {
+                if (!isWasmTargetAdded()) {
+                    askYesNoQuestionAndRunFunction(addWasm32TragetQuestion, addWasm32Target);
+                }
+                return false;
+            } else {
+                const isAnswerPositive = askYesNoQuestionAndRunFunction(installRustupQuestion, installRustup);
+                askYesNoQuestionAndRunFunction(addWasm32TragetQuestion, addWasm32Target);
+                return isAnswerPositive;
+            }
+        } else {
```

Comment:
nit: I feel it can be a bit cleaner with early returns:

```js
if (!isUnix) {
  askYesNoQuestionAndRunFunction(rustupAndWasm32WindowsInstalationInstructions);
  return false;
}

if (!isRustupInstalled()) {
  const isAnswerPositive = askYesNoQuestionAndRunFunction(installRustupQuestion, installRustup);
  askYesNoQuestionAndRunFunction(addWasm32TragetQuestion, addWasm32Target);
  return isAnswerPositive;
}

...

```

basically this pattern allows to reduce nesting and highlight what's a "default" path

can be useful in other instances of `if/else` as well, not just here